### PR TITLE
World Bank data loader

### LINKED ISSUE #19: Download script for WB data
Write a script that will:

1. Check if `data/world_bank.csv` exists
2. If not, download, preprocess and save
3. If yes, load it 

`world_bank.csv` should contain the time series identified in #5, for all countries possible. Annual frequency is fine, since that's what we have in EM-DAT anyway. Preprocessing should put the dataset into a panel format, with (date, ISO3) index and time series names on the columns. Rename the time series to be human-understandable. Add a dictionary global variable that maps between the WB database names and our human-readable names.

### PR DESCRIPTION
**Pull request by CamSalda**
Creates the script to download the World Bank data file and add the variable dictionary to the constant_variables file.

I tried to download all the data from a single URL, but was not possible. Therefore the scripts download each file and then it merges them.
Closes #19

### CODE REVIEW

**Comment by jessegrabowski** [comment]
Did you try to use `pandas_dataloader` for this? Example [here](https://github.com/jessegrabowski/psme_macro_2023/blob/main/Notebooks/TD%2010.%20Project%20Walkthough.ipynb) in the "data preparation" section

File: laos_gggi/world_bank_data_loader.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                "EN.POP.DNST": "population_density",
+                "NY.GDP.PCAP.KD": "gdp_per_cap",
+            },
+            inplace=True,
+        )
+        wb_df = wb_df.set_index(["country_code", "year"]).sort_index()
+
+    else:
+        # Importing data
+        wb_df = wb.download(
```
Comment (jessegrabowski) [inline laos_gggi/world_bank_data_loader.py:38]: This is the branch where the file exists, so you should load from disk rather than downloading

**Review by jessegrabowski — CHANGES_REQUESTED** [review]
Check that the `if` and `else` branches are correct. Otherwise just a small optimization to the country renaming.

Make sure you run the code in a notebook locally to make sure it works

File: laos_gggi/world_bank_data_loader.py (hunk 1 above)
Comment (CamSalda) [reply laos_gggi/world_bank_data_loader.py:38]: Addressed!

File: laos_gggi/world_bank_data_loader.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        wb_df.rename(
+            columns={
+                "EN.POP.DNST": "population_density",
+                "NY.GDP.PCAP.KD": "gdp_per_cap",
+            },
+            inplace=True,
+        )
+        wb_df = wb_df.set_index(["country_code", "year"]).sort_index()
+
+    return (wb_df, wb_df.to_csv(path_to_wb_data))
```
[the remark this answers is not in the document]
Comment (CamSalda) [reply laos_gggi/world_bank_data_loader.py:60]: Done!

File: laos_gggi/world_bank_data_loader.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            indicator=["EN.POP.DNST", "NY.GDP.PCAP.KD"],
+            country=COUNTRIES_ISO,
+            start="1900",
+            end=None,
+        )
+        wb_df.reset_index(inplace=True)
+        # Adding country code
+        wb_df["country_code"] = np.NaN
+
+        for x in wb_df.index.values:
```
[the remark this answers is not in the document]
Comment (CamSalda) [reply laos_gggi/world_bank_data_loader.py:23]: Done!

**Comment by CamSalda** [comment]
> Check that the `if` and `else` branches are correct. Otherwise just a small optimization to the country renaming.
> 
> Make sure you run the code in a notebook locally to make sure it works

Also done.

File: laos_gggi/world_bank_data_loader.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            columns={
+                "EN.POP.DNST": "population_density",
+                "NY.GDP.PCAP.KD": "gdp_per_cap",
+            },
+            inplace=True,
+        )
+        wb_df = wb_df.set_index(["country_code", "year"]).sort_index()
+        wb_df.to_csv(path_to_wb_data)
+
+    if not exists(path_to_wb_data):
```
Comment (jessegrabowski) [inline laos_gggi/world_bank_data_loader.py:34]: How is this different than the first `if` branch?

File: laos_gggi/world_bank_data_loader.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+from const_vars import COUNTRIES_ISO, ISO_DICTIONARY
+
+
+def download_wb_data(output_path="../data"):
+    path_to_wb_data = output_path + "/world_bank.csv"
+    if not exists(output_path):
+        os.makedirs(output_path)
+        # Importing data
+        wb_df = wb.download(
+            indicator=["EN.POP.DNST", "NY.GDP.PCAP.KD"],
```
Comment (jessegrabowski) [inline laos_gggi/world_bank_data_loader.py:14]: add a global variable `WB_INDICATORS = ["EN.POP.DNST", "NY.GDP.PCAP.KD"]` and use it here

File: laos_gggi/world_bank_data_loader.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            country=COUNTRIES_ISO,
+            start="1900",
+            end=None,
+        )
+        wb_df.reset_index(inplace=True)
+        # Adding country code
+        wb_df["country_code"] = wb_df["country"].apply(ISO_DICTIONARY.get)
+        # Formatting data
+        wb_df = wb_df[["country_code", "year", "NY.GDP.PCAP.KD", "EN.POP.DNST"]]
```
Comment (jessegrabowski) [inline laos_gggi/world_bank_data_loader.py:23]: `wb_df = wb_df[["country_code", "year"] + WB_INDICATORS]`

File: laos_gggi/world_bank_data_loader.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        wb_df.rename(
+            columns={
+                "EN.POP.DNST": "population_density",
+                "NY.GDP.PCAP.KD": "gdp_per_cap",
```
Comment (jessegrabowski) [inline laos_gggi/world_bank_data_loader.py:27]: make a `WB_RENAME_DICT = {"EN.POP.DNST": "population_density", "NY.GDP.PCAP.KD": "gdp_per_cap"}` global.

Basically anything that: 
1. You are using more than once, or;
2. We would maybe like to be able to change later (add/remove things)

Should be a global constant variable, like a configuration setting.

**Review by jessegrabowski — CHANGES_REQUESTED** [review]
There seems to be a duplicated `if` branch

File: laos_gggi/world_bank_data_loader.py (hunk 4 above)
Comment (CamSalda) [reply laos_gggi/world_bank_data_loader.py:34]: Basically, the first one checks if the "data" folder exists. The second one checks if the file exists. I just changed the name to "folder_path" to make it clearer.

File: laos_gggi/world_bank_data_loader.py (hunk 5 above)
Comment (CamSalda) [reply laos_gggi/world_bank_data_loader.py:14]: Done!

File: laos_gggi/world_bank_data_loader.py (hunk 7 above)
Comment (CamSalda) [reply laos_gggi/world_bank_data_loader.py:27]: Done!

Comment (CamSalda) [reply laos_gggi/world_bank_data_loader.py]: I just pushed a new version with all these changes

File: laos_gggi/world_bank_data_loader.py (hunk 6 above)
Comment (CamSalda) [reply laos_gggi/world_bank_data_loader.py]: Done! I just pushed a new version with all these observations

**Review by jessegrabowski — APPROVED** [review]
Looks good!

**Pull request merged**